src/core/src/op/unique.cpp

```diff
@@ -0,0 +1,123 @@
+                        const element::Type& index_element_type)
+    : op::Op{{data, axis}},
+      m_sorted{sorted},
+      m_index_element_type{index_element_type} {
+    constructor_validate_and_infer_types();
+}
+
+bool op::v10::Unique::visit_attributes(AttributeVisitor& visitor) {
+    OV_OP_SCOPE(v10_Unique_visit_attributes);
+    visitor.on_attribute("sorted", m_sorted);
+    visitor.on_attribute("index_element_type", m_index_element_type);
+    return true;
+}
+
+void op::v10::Unique::validate_and_infer_types() {
+    OV_OP_SCOPE(v10_Unique_validate_and_infer_types);
+    NODE_VALIDATION_CHECK(this,
+                          m_index_element_type == element::i32 || m_index_element_type == element::i64,
+                          "The element type of the outputs containing indices can only be set to i32 or i64");
+
+    const auto& input_shape = get_input_partial_shape(0);
+    std::vector<PartialShape> output_shapes(4);
+
+    int64_t input_tensor_capacity = -1;
+    if (input_shape.is_static()) {
+        input_tensor_capacity = static_cast<int64_t>(shape_size(input_shape.to_shape()));
+    }
+
+    output_shapes[0] = PartialShape::dynamic();
+    output_shapes[1] =
+        input_tensor_capacity > 0 ? PartialShape{{1, input_tensor_capacity}} : PartialShape{{Dimension::dynamic()}};
+    output_shapes[2] = output_shapes[1];
+    output_shapes[3] = output_shapes[1];
+
+    if (get_input_size() == 2) {
+        NODE_VALIDATION_CHECK(
+            this,
+            get_input_element_type(1) == element::i32 || get_input_element_type(1) == element::i64,
+            "The allowed element types of the 'axis' input tensor of the Unique operator are i32 and i64.");
+
+        NODE_VALIDATION_CHECK(
+            this,
+            get_input_partial_shape(1) == Shape{} || get_input_partial_shape(1) == Shape{1},
+            "The 'axis' input tensor of the Unique operator must be a scalar or 1D tensor with 1 element.");
+
+        NODE_VALIDATION_CHECK(this,
+                              ov::op::util::is_constant(input_value(1).get_node()),
+                              "The 'axis' input of the Unique operator must be connected to a Constant.");
+        const int64_t axis =
+            extract_axis(std::dynamic_pointer_cast<op::v0::Constant>(input_value(1).get_node_shared_ptr()));
+
+        if (input_shape.rank().is_static()) {
+            const auto normalized_axis = ngraph::normalize_axis(this, axis, input_shape.rank());
+            const auto dim_at_axis = input_shape[normalized_axis];
+            if (dim_at_axis.is_static()) {
+                auto output_shape = input_shape;
+                output_shape[normalized_axis] = Dimension{1, dim_at_axis.get_length()};
+                output_shapes[0] = output_shape;
+            } else {
+                auto output_shape = input_shape;
+                output_shape[normalized_axis] = Dimension::dynamic();
+                output_shapes[0] = output_shape;
+            }
```

Comment:
This could be unified for static and dynamic dimension:
```suggestion
                auto output_shape = input_shape;
                output_shape[normalized_axis] = Dimension{0, dim_at_axis.get_max_length()};
                output_shapes[0] = output_shape;
```
Corner case, but it's possible to get `0` dimension in the input.

Reply:
This unified approach produces a dimension with upper bound equal to max length every time. For static shapes this is incorrect because the upper bound should not be that large.

Reply:
Was it tested? As far I know, `get_max_length()` should return `upper_bound` for dynamic and static. 
For static dimension, `upper_bound` is equal to the dimension value and `lower == upper`

Reply:
I will double check and confirm.

Reply:
I've unified the code but in a slightly different way. I'm not using zero as lower bound because for most of the cases this would not be true.